mne/viz/backends/_ipyvolume.py

```diff
@@ -0,0 +1,366 @@
+
+
+@copy_base_doc_to_subclass_doc
+class _Renderer(_BaseRenderer):
+    """Class managing rendering scene created with ipyvolume.
+
+    Attributes
+    ----------
+    plotter: ipyvolume.widgets.Figure
+        The scene handler.
+    """
+
+    def __init__(self, fig=None, size=(600, 600), bgcolor=(0., 0., 0.),
+                 name="Ipyvolume Scene", show=False):
+        """Set up the scene.
+
+        Parameters
+        ----------
+        fig: ipyvolume.widgets.Figure
+            The scene handler.
+        size: tuple
+            The dimensions of the context window: (width, height).
+        bgcolor: tuple
+            The color definition of the background: (red, green, blue).
+        name: str | None
+            The name of the scene.
+        """
+        self.off_screen = False
+        self.name = name
+
+        if fig is None:
+            fig_w, fig_h = size
+            self.plotter = ipv.figure(width=fig_w, height=fig_h)
+            self.plotter.animation = 0
+            ipv.style.box_off()
+            ipv.style.axes_off()
+            bgcolor = tuple(int(c) for c in bgcolor)
+            ipv.style.background_color('#%02x%02x%02x' % bgcolor)
+        else:
+            self.plotter = ipv.figure(key=fig)
+
+    def scene(self):
+        return self.plotter
+
+    def set_interactive(self):
+        pass
+
+    def mesh(self, x, y, z, triangles, color, opacity=1.0, shading=False,
+             backface_culling=False, **kwargs):
+        # opacity for overlays will be provided as part of color
+        color = _color2rgba(color, opacity)
+        mesh = ipv.plot_trisurf(x, y, z, triangles=triangles, color=color)
+        _add_transperent_material(mesh)
+        ipv.squarelim()
+
+        return mesh
+
+    def contour(self, surface, scalars, contours, line_width=1.0, opacity=1.0,
+                vmin=None, vmax=None, colormap=None):
+        from matplotlib import cm
+        from matplotlib.colors import ListedColormap
+        from ipyvolume.pylab import plot
+
+        vertices = surface['rr']
+        tris = surface['tris']
+
+        if colormap is None:
+            cmap = cm.get_cmap('coolwarm')
+        else:
+            cmap = ListedColormap(colormap / 255.0)
+
+        if isinstance(contours, int):
+            levels = np.linspace(vmin, vmax, num=contours)
+        else:
+            levels = np.array(contours)
+
+        if scalars is not None:
+            if vmin is None:
+                vmin = min(scalars)
+            if vmax is None:
+                vmax = max(scalars)
+            nscalars = (scalars - vmin) / (vmax - vmin)
+            color = cmap(nscalars)
+        else:
+            color = np.append(color, opacity)
+
+        verts, faces, _, _ = _isoline(vertices=vertices, tris=tris,
+                                      vertex_data=scalars,
+                                      levels=levels)
+
+        x = verts[:, 0]
+        y = verts[:, 1]
+        z = verts[:, 2]
```

Comment:
`x, y, z = verts` (if it is actually (n, 3), otherwise it's like `x, y, z = verts[:, :3]`